### ISSUE TITLE
"No incoming connections" spam in my console

### ISSUE THREAD

**Issue by SgtTheDarknezz** [aeonix/aeon#64]
So, after updating to point release 6, I've been getting my console spammed with this message:

2018-10-11 03:39:04.983  [P2P1]  WARN    global  src/p2p/net_node.inl:1310       No incoming connections - check firewalls/routers allow port 11180

I've allowed the port in my modem+iptables, i am running ubuntu 16.04, blockchain still updates fine, seems that transactions work fine too.

**Comment by shigutso** [comment]
yes this happens on my daemon too

**Comment by BigslimVdub** [comment]
are you running the daemon as is or adding additional options to the start of the daemon such as this -  `--restricted-rpc`

or 

`--data-dir=d:\aeon`

I too get some of these lines right after start but after running the daemon for a bit and getting some in connections, I no longer get the warns. I do NOT have my firewall ports 11180 or 11181 open (Technically you should have 11180 open to allow incoming connections)

As long as you have in connections, I would Ignore it. Type in `status` and see how many in connections you have. Usually I get 4-5 in connections running the daemon 24/7.

Also these can be limited by `in_peers 10` to limit in peers to 10 (change 10 to however many you want)

**Comment by camthegeek** [comment]
I ended up opening firewall port 11180 (for p2p) on one set of servers (4x nodes dealing with my pool) and the problem was solved.

Additionally, on a 2nd server, I increased `in_peers` to 25 and the warnings died off.

**Comment by BigslimVdub** [comment]
Possibly @stoffu or @iamsmooth can comment if the daemon is expecting a specific number of in connections and if not detected, throws this warning. As stated, when I get 4-5 in connections I get no more warnings but they do show up until the node sees a few incoming connections.

**Comment by stoffu** [comment]
This message is quite literal: you have no incoming connections, meaning peers can't download blocks/txes from you. If you see something like `8(out)+0(in) connections` with the `status` command, you have no incoming connections. You can allow incoming connections by opening port 11180.

**Comment by BigslimVdub** [comment]
Was anything changed in the update from v.0.12.5.0 to v.0.12.6.0 for this, because I have never seen this warning message in over a year even with old 9.0 software pre-rebase and I personally have never had an open port either? 
If nothing was changed I would say we should close this bug as daemon is acting as it should.

**Comment by stoffu** [comment]
It was added when the upstream patches were merged in August (which was after v0.12.5): https://github.com/aeonix/aeon/commit/65eacd6cecea206efd77c8984886830ed42ce38d#diff-c02157891426370c008d7076b723128c

**Comment by BigslimVdub** [comment]
Well, the above patch would be the reason I never saw this warning before!

Since Aeon does not have a large node network, would increasing the check time to 60 minutes (3600 seconds) be acceptable before showing the warning to possibly help reduce this warning if users daemons have issues finding nodes to connect?

Is this the correct change for that addition?

```
 epee::math_helper::once_a_time_seconds<3600, false> m_incoming_connections_interval;

```

**Comment by stoffu** [comment]
@BigslimVdub 
That is the correct place to change.

> if users daemons have issues finding nodes to connect

This warning will always appear when port 11180 is closed, regardless of the user's internet connection quality. If this feels spammy, extending the interval from 15 minutes to 60 minutes will make it only 1/4 less spammy. I think this warning is useful because it encourages people to open port 11180 which is good for the network. Those who feel this to be too spammy can run the daemon with `--hide-my-port`.

**Comment by iamsmooth** [comment]
Fixed by #81
